libs/application/template-api-modules/src/lib/modules/templates/reference-template/reference-template.service.ts

```diff
@@ -2,18 +2,50 @@ import { Injectable } from '@nestjs/common'
 
 import { SharedTemplateApiService } from '../../shared'
 import { TemplateApiModuleActionProps } from '../../../types'
-
+import { getValueViaPath } from '@island.is/application/core'
 import {
   generateApplicationApprovedEmail,
   generateAssignApplicationEmail,
 } from './emailGenerators'
+import { ApplicationTypes } from '@island.is/application/types'
+import { BaseTemplateApiService } from '../../base-template-api.service'
 
 const TWO_HOURS_IN_SECONDS = 2 * 60 * 60
 @Injectable()
-export class ReferenceTemplateService {
+export class ReferenceTemplateService extends BaseTemplateApiService {
   constructor(
     private readonly sharedTemplateAPIService: SharedTemplateApiService,
-  ) {}
+  ) {
+    super(ApplicationTypes.EXAMPLE)
+  }
+
+  async getReferenceData({ application }: TemplateApiModuleActionProps) {
+    await new Promise((resolve) => setTimeout(resolve, 2000))
+
+    const name = getValueViaPath(
+      application.externalData,
+      'nationalRegistry.data.name',
+    ) as string
+
```

Comment:
Would be nice to see the reference template throw a `TemplateApiError`, for reference..